plaso/parsers/google_logging.py

```diff
@@ -0,0 +1,264 @@
+      _PYPARSING_COMPONENTS['hour'] + pyparsing.Suppress(':') +
+      _PYPARSING_COMPONENTS['minute'] + pyparsing.Suppress(':') +
+      _PYPARSING_COMPONENTS['second'] + pyparsing.Optional(
+          pyparsing.Suppress('.') +
+          _PYPARSING_COMPONENTS['microsecond']))
+
+  # Grammar for individual log event lines.
+  _LOG_LINE = (
+      _PYPARSING_COMPONENTS['priority'] + _PYPARSING_COMPONENTS['date']
+      + _PYPARSING_COMPONENTS['thread_id'] +
+      _PYPARSING_COMPONENTS['file_name'] + pyparsing.Suppress(':') +
+      _PYPARSING_COMPONENTS['line_number'] + pyparsing.Suppress('] ') +
+      _PYPARSING_COMPONENTS['message'] + pyparsing.lineEnd())
+
+  # Grammar for the log file greeting.
+  _GREETING = (
+      _PYPARSING_COMPONENTS['year'] + pyparsing.Suppress('/') +
+      _PYPARSING_COMPONENTS['month_number'] + pyparsing.Suppress('/') +
+      _PYPARSING_COMPONENTS['day'] + _PYPARSING_COMPONENTS['hour'] +
+      pyparsing.Suppress(':') + _PYPARSING_COMPONENTS['minute'] +
+      pyparsing.Suppress(':') + _PYPARSING_COMPONENTS['second'] +
+      pyparsing.Regex('.*?(?=($|\n[IWEF][0-9]{4}))', re.DOTALL) +
+      pyparsing.lineEnd())
+
+  _GREETING_START = 'Log file created at: '
+
+  # Our initial buffer length is the length of the string we verify with.
+  _INITIAL_BUFFER_SIZE = len(_GREETING_START)
+
+  # Once we're sure we're reading a valid file, we'll use a larger read buffer.
+  _RUNNING_BUFFER_SIZE = 5120
+
+  # Order is important here, as the structures are checked against each line
+  # sequentially, so we put the most common first, and the most expensive
+  # last.
+  LINE_STRUCTURES = [
+      ('log_entry', _LOG_LINE),
+      ('greeting_start', pyparsing.Literal(_GREETING_START)),
+      ('greeting', _GREETING)]
+
+  _SUPPORTED_KEYS = frozenset([key for key, _ in LINE_STRUCTURES])
+
+  def __init__(self):
+    """Initialize a parser object."""
+    super(GoogleLogParser, self).__init__()
+
+    # Set the size of the file we need to read to verify it.
+    self._buffer_size = self._INITIAL_BUFFER_SIZE
+    self._maximum_year = 0
+    # The year to use for events. The initial year is stored in the log file
+    # greeting.
+    self._year_use = 0
+    # The month the last observed event occurred.
+    self._last_month = 0
+
+  def _UpdateYear(self, mediator, month):
+    """Updates the year to use for events, based on last observed month.
+
+    Args:
+      mediator (ParserMediator): mediates the interactions between
+          parsers and other components, such as storage and abort signals.
+      month (int): month observed by the parser, where January is 1.
+    """
+    if not self._year_use:
+      self._year_use = mediator.GetEstimatedYear()
+    if not self._maximum_year:
+      self._maximum_year = mediator.GetLatestYear()
+
+    if not self._last_month:
+      self._last_month = month
+      return
+
+    # TODO: Check whether out of order events are possible
+    if self._last_month > (month + 1):
```

Comment:
Why `month + 1` ? I assume `self._last_month` contains a value from 1 upto and including 12
and month as well. So if the logfile would jump from Dec 2018 to Nov 2019 the year would stay the same?

Reply:
Thats correct, this is the same logic as used in the syslog parser, where events can sometimes occur out of order. In your example, this code assumes that if an event from November follows an event from December 2018, that the event occurred in 2018, not 2019.

Reply:
Ack, added to https://github.com/log2timeline/plaso/issues/434